yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
                 "'{}' is unrecognized ranking method".format(algorithm)
             )
 
+        # Extract matrix from dataframe if necessary
+        if is_dataframe(X):
```

Comment:
My interpretation of the original intent for the `features` argument was that it was to be used to pass in labels for display on the plot, not for filtering or re-ordering columns.  However, if you'd like to add that functionality, we could do it.  

The only issue that I see is that this leads to an asymmetry in the treatment of numpy arrays vs. dataframes.  In the dataframe case, passing a `features` argument will potentially reorder the columns, while in the numpy case the same `features` argument will leave the order the same but just label the columns.  So, the same data and same `features` argument will give you different results depending on which type the data is.  Might be confusing, but maybe that's something we can live with?  Just let me know what you think and we can go either way.